Multiple occurences of unresolved symbols from snappy in leveldb

### Affected Branch

trunk

### Basic Diagnostics

- [X] I've pulled the latest changes on the affected branch and the issue is still present.

- [ ] The issue is reproducible in docker

### Description

Only example of error message from build. Multiple unresolved symbols from snappy library when building on Arch Linux.
```
[ 73%] Building CXX object src/uhs/twophase/sentinel_2pc/CMakeFiles/sentineld-2pc.dir/sentineld_2pc.o
/usr/bin/ld: /usr/local/lib/libleveldb.a(table_builder.cc.o): in function `leveldb::port::Snappy_Compress(char const*, unsigned long, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >*)':
/home/prauat/projects/opencbdc-tx/leveldb-1.22-Debug/port/port_stdcxx.h:91: undefined reference to `snappy::MaxCompressedLength(unsigned long)'
/usr/bin/ld: /home/prauat/projects/opencbdc-tx/leveldb-1.22-Debug/port/port_stdcxx.h:93: undefined reference to `snappy::RawCompress(char const*, unsigned long, char*, unsigned long*)'
/usr/bin/ld: /usr/local/lib/libleveldb.a(format.cc.o): in function `leveldb::port::Snappy_GetUncompressedLength(char const*, unsigned long, unsigned long*)':
/home/prauat/projects/opencbdc-tx/leveldb-1.22-Debug/port/port_stdcxx.h:109: undefined reference to `snappy::GetUncompressedLength(char const*, unsigned long, unsigned long*)'
/usr/bin/ld: /usr/local/lib/libleveldb.a(format.cc.o): in function `leveldb::port::Snappy_Uncompress(char const*, unsigned long, char*)':
/home/prauat/projects/opencbdc-tx/leveldb-1.22-Debug/port/port_stdcxx.h:121: undefined reference to `snappy::RawUncompress(char const*, unsigned long, char*)'
collect2: error: ld returned 1 exit status
make[2]: *** [src/uhs/atomizer/archiver/CMakeFiles/archiverd.dir/build.make:108: src/uhs/atomizer/archiver/archiverd] Błąd 1
make[1]: *** [CMakeFiles/Makefile2:774: src/uhs/atomizer/archiver/CMakeFiles/archiverd.dir/all] Błąd 2
make[1]: *** Oczekiwanie na niezakończone zadania....
/usr/bin/ld: /usr/local/lib/libleveldb.a(table_builder.cc.o): in function `leveldb::port::Snappy_Compress(char const*, unsigned long, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >*)':
/home/prauat/projects/opencbdc-tx/leveldb-1.22-Debug/port/port_stdcxx.h:91: undefined reference to `snappy::MaxCompressedLength(unsigned long)'
/usr/bin/ld: /home/prauat/projects/opencbdc-tx/leveldb-1.22-Debug/port/port_stdcxx.h:93: undefined reference to `snappy::RawCompress(char const*, unsigned long, char*, unsigned long*)'
/usr/bin/ld: /usr/local/lib/libleveldb.a(format.cc.o): in function `leveldb::port::Snappy_GetUncompressedLength(char const*, unsigned long, unsigned long*)':
/home/prauat/projects/opencbdc-tx/leveldb-1.22-Debug/port/port_stdcxx.h:109: undefined reference to `snappy::GetUncompressedLength(char const*, unsigned long, unsigned long*)'
/usr/bin/ld: /usr/local/lib/libleveldb.a(format.cc.o): in function `leveldb::port::Snappy_Uncompress(char const*, unsigned long, char*)':
/home/prauat/projects/opencbdc-tx/leveldb-1.22-Debug/port/port_stdcxx.h:121: undefined reference to `snappy::RawUncompress(char const*, unsigned long, char*)'
collect2: error: ld returned 1 exit status
make[2]: *** [src/uhs/twophase/locking_shard/CMakeFiles/locking-shardd.dir/build.make:108: src/uhs/twophase/locking_shard/locking-shardd] Błąd 1
make[1]: *** [CMakeFiles/Makefile2:1281: src/uhs/twophase/locking_shard/CMakeFiles/locking-shardd.dir/all] Błąd 2
```
In order to reproduce the issue, follow these steps:

1. scripts/configure.sh
2. scripts/build.sh

Well it looks like snappy should be added to CMakeList.txt in multiple files using check_library_exists() and target_link_libraries().

Proposed changes can be viewed [arch-linux-snappy](https://github.com/mit-dci/opencbdc-tx/compare/trunk...pr4u4t:opencbdc-tx:arch-linux-snappy). Let me know if pull request is desired.  

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -32,6 +32,7 @@ find_library(LEVELDB_LIBRARY leveldb REQUIRED)
 find_library(NURAFT_LIBRARY nuraft REQUIRED)
 find_library(GTEST_LIBRARY gtest REQUIRED)
 find_library(GTEST_MAIN_LIBRARY gtest_main REQUIRED)
+find_library(SNAPPY_LIBRARY snappy REQUIRED)
 
 set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fno-rtti")
 

--- a/scripts/configure.sh
+++ b/scripts/configure.sh
@@ -26,7 +26,7 @@ elif [[ "$OSTYPE" == "darwin"* ]]; then
   CPUS=$(sysctl -n hw.ncpu)
   # ensure development environment is set correctly for clang
   $SUDO xcode-select -switch /Library/Developer/CommandLineTools
-  brew install llvm@14 googletest lcov make wget cmake
+  brew install llvm@14 googletest lcov make wget cmake snappy
   CLANG_TIDY=/usr/local/bin/clang-tidy
   if [ ! -L "$CLANG_TIDY" ]; then
     $SUDO ln -s $(brew --prefix)/opt/llvm@14/bin/clang-tidy /usr/local/bin/clang-tidy
@@ -39,7 +39,7 @@ fi
 
 if [[ "$OSTYPE" == "linux-gnu"* ]]; then
   apt update
-  apt install -y build-essential wget cmake libgtest-dev lcov git software-properties-common rsync
+  apt install -y build-essential wget cmake libgtest-dev lcov git software-properties-common rsync libsnappy-dev
 
   wget -O - https://apt.llvm.org/llvm-snapshot.gpg.key | $SUDO apt-key add -
   $SUDO add-apt-repository "deb http://apt.llvm.org/focal/ llvm-toolchain-focal-14 main"

--- a/src/uhs/atomizer/archiver/CMakeLists.txt
+++ b/src/uhs/atomizer/archiver/CMakeLists.txt
@@ -15,4 +15,5 @@ target_link_libraries(archiverd archiver
                                 ${LEVELDB_LIBRARY}
                                 ${NURAFT_LIBRARY}
                                 secp256k1
-                                ${CMAKE_THREAD_LIBS_INIT})
+                                ${CMAKE_THREAD_LIBS_INIT}
+                                snappy)

--- a/src/uhs/atomizer/atomizer/CMakeLists.txt
+++ b/src/uhs/atomizer/atomizer/CMakeLists.txt
@@ -23,4 +23,5 @@ target_link_libraries(atomizer-raftd atomizer_raft
                                      ${NURAFT_LIBRARY}
                                      ${LEVELDB_LIBRARY}
                                      secp256k1
-                                     ${CMAKE_THREAD_LIBS_INIT})
+                                     ${CMAKE_THREAD_LIBS_INIT}
+                                     snappy)

--- a/src/uhs/atomizer/shard/CMakeLists.txt
+++ b/src/uhs/atomizer/shard/CMakeLists.txt
@@ -16,4 +16,5 @@ target_link_libraries(shardd shard
                              ${LEVELDB_LIBRARY}
                              ${NURAFT_LIBRARY}
                              secp256k1
-                             ${CMAKE_THREAD_LIBS_INIT})
+                             ${CMAKE_THREAD_LIBS_INIT}
+                             snappy)

--- a/src/uhs/atomizer/watchtower/CMakeLists.txt
+++ b/src/uhs/atomizer/watchtower/CMakeLists.txt
@@ -21,4 +21,5 @@ target_link_libraries(watchtowerd watchtower
                                   crypto
                                   ${NURAFT_LIBRARY}
                                   secp256k1
-                                  ${CMAKE_THREAD_LIBS_INIT})
+                                  ${CMAKE_THREAD_LIBS_INIT}
+                                  snappy)

--- a/src/uhs/twophase/coordinator/CMakeLists.txt
+++ b/src/uhs/twophase/coordinator/CMakeLists.txt
@@ -20,4 +20,5 @@ target_link_libraries(coordinatord coordinator
                                    ${NURAFT_LIBRARY}
                                    ${LEVELDB_LIBRARY}
                                    secp256k1
-                                   ${CMAKE_THREAD_LIBS_INIT})
+                                   ${CMAKE_THREAD_LIBS_INIT}
+                                   snappy)

--- a/src/uhs/twophase/locking_shard/CMakeLists.txt
+++ b/src/uhs/twophase/locking_shard/CMakeLists.txt
@@ -22,4 +22,5 @@ target_link_libraries(locking-shardd locking_shard
                                      secp256k1
                                      ${NURAFT_LIBRARY}
                                      ${LEVELDB_LIBRARY}
-                                     ${CMAKE_THREAD_LIBS_INIT})
+                                     ${CMAKE_THREAD_LIBS_INIT}
+                                     snappy)

--- a/tests/integration/CMakeLists.txt
+++ b/tests/integration/CMakeLists.txt
@@ -37,4 +37,5 @@ target_link_libraries(run_integration_tests ${GTEST_LIBRARY}
                                             secp256k1
                                             ${LEVELDB_LIBRARY}
                                             ${NURAFT_LIBRARY}
-                                            ${CMAKE_THREAD_LIBS_INIT})
+                                            ${CMAKE_THREAD_LIBS_INIT}
+                                            snappy)

--- a/tests/unit/CMakeLists.txt
+++ b/tests/unit/CMakeLists.txt
@@ -55,4 +55,5 @@ target_link_libraries(run_unit_tests ${GTEST_LIBRARY}
                                      secp256k1
                                      ${LEVELDB_LIBRARY}
                                      ${NURAFT_LIBRARY}
-                                     ${CMAKE_THREAD_LIBS_INIT})
+                                     ${CMAKE_THREAD_LIBS_INIT}
+                                     snappy)

--- a/tools/shard-seeder/CMakeLists.txt
+++ b/tools/shard-seeder/CMakeLists.txt
@@ -12,4 +12,5 @@ target_link_libraries(shard-seeder transaction
                                    crypto
                                    ${LEVELDB_LIBRARY}
                                    ${CMAKE_THREAD_LIBS_INIT}
-                                   secp256k1)
+                                   secp256k1
+                                   snappy)

